bigdata-telemetry/src/main/java/io/tidb/bigdata/telemetry/TeleMsg.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 TiDB Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.tidb.bigdata.telemetry;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Telemetry message. */
+public abstract class TeleMsg {
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  public String trackId;
+  public long time;
+  public String subName;
+  public Map<String, Object> hardware;
```

Comment:
Why use Object

Reply:
Because the value in the map may be int, String, and Map.